internal/handlers/boot_artifacts.go

```diff
@@ -64,7 +70,16 @@ func (b *BootArtifactsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 	}
 
 	isoFileName := b.ImageStore.PathForParams(imagestore.ImageTypeFull, version, arch)
-	fileReader, err := isoeditor.GetFileFromISO(isoFileName, fmt.Sprintf("/images/pxeboot/%s", artifact))
+	// in case of ins file adjust file path
+	file_path := ""
+
+	if strings.HasSuffix(artifact, ".ins") {
+		file_path = "/"
+	} else {
+		file_path = "/images/pxeboot/"
+	}
+
+	fileReader, err := isoeditor.GetFileFromISO(isoFileName, fmt.Sprintf("%s%s", file_path, artifact))
```

Comment:
```suggestion
	file_path := fmt.Sprintf("/images/pxeboot/%s", artifact)
	if artifact == "generic.ins" {
		file_path = "/generic.ins"
	}

	fileReader, err := isoeditor.GetFileFromISO(isoFileName, file_path)
```

Also maybe we should make `generic.ins` a constant?

Reply:
Making generic.ins a constant is not necessary (I think). Pls check my alternative code ..